cpp/src/arrow/extension_type.cc

```diff
@@ -146,14 +147,17 @@ static void CreateGlobalRegistry() {
   g_registry = std::make_shared<ExtensionTypeRegistryImpl>();
 
 #ifdef ARROW_JSON
-  // Register canonical extension types
-  auto fst_ext_type =
-      checked_pointer_cast<ExtensionType>(extension::fixed_shape_tensor(int64(), {}));
-  ARROW_CHECK_OK(g_registry->RegisterType(fst_ext_type));
-
-  auto bool8_ext_type = checked_pointer_cast<ExtensionType>(extension::bool8());
-  ARROW_CHECK_OK(g_registry->RegisterType(bool8_ext_type));
+  std::vector<std::shared_ptr<DataType>> ext_types{
+      extension::bool8(), extension::fixed_shape_tensor(int64(), {}), extension::uuid()};
+#else
+  std::vector<std::shared_ptr<DataType>> ext_types{extension::bool8(), extension::uuid()};
 #endif
```

Comment:
Suggestion to make this neater and less repetitive:
```c++
  std::vector<std::shared_ptr<DataType>> ext_types{extension::bool8(), extension::uuid()};
#ifdef ARROW_JSON
  ext_types.push_back(extension::fixed_shape_tensor(int64(), {}));
#endif
```

Reply:
Changed.